sqlalchemy_hana/dialect.py

```diff
@@ -531,15 +535,22 @@ class HANAHDBCLIDialect(default.DefaultDialect):
     isolation_level = None
     default_schema_name: str  # this is always set for us
 
+    _json_deserializer = None
+    _json_serializer = None
+
     def __init__(
         self,
         isolation_level: str | None = None,
         use_native_boolean: bool = True,
+        json_serializer: Optional[Callable[[Any], str]] = None,
+        json_deserializer: Optional[Callable[[str], Any]] = None,
```

Comment:
create_engine already takes these arguments: https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.create_engine.params.json_serializer

Reply:
Without explicitly adding them as instance attributes, tests will fail, as then on sqlalchemy level they check if its set:
https://github.com/sqlalchemy/sqlalchemy/blob/83bd285e7788ee19348355988bc3874824cebbec/lib/sqlalchemy/sql/sqltypes.py#L2686
which otherwise leads to an AttributeError. This allows us to a)  set a default implementation and b) honor the engine parameters. Check also reference implementation, e.g., postgres: https://github.com/sqlalchemy/sqlalchemy/blob/83bd285e7788ee19348355988bc3874824cebbec/lib/sqlalchemy/dialects/postgresql/base.py#L3112